python/ray/data/tests/test_dataset_numpy.py

```diff
@@ -122,8 +122,8 @@ def test_numpy_roundtrip(ray_start_regular_shared, fs, data_path):
     ds.write_numpy(data_path, filesystem=fs)
     ds = ray.data.read_numpy(data_path, filesystem=fs)
     assert str(ds) == (
-        "Dataset(\n\tnum_blocks=2,\n\tnum_rows=?,"
-        "\n\tschema={__value__: ArrowTensorType(shape=(1,), dtype=int64)}\n)"
+        "Dataset(\n    num_blocks=2,\n    num_rows=?,"
+        "\n    schema={__value__: ArrowTensorType(shape=(1,), dtype=int64)}\n)"
```

Comment:
Could do this later, but we probably shouldn't assert against `repr`s. There's a high maintenance cost. `repr`s can change between versions, and we'd have to update the tests each time.

Reply:
would the best alternative be to directly inspect the `ds` object's attributes, e.g. `ds.num_blocks == 2`, etc?

Reply:
Not sure if we'd want to test `num_blocks` for this test in particular, but yeah, I think directly inspecting makes sense. Like, we could test the schema with `Dataset.schema` and the data values with `Dataset.takeall` or something.

Reply:
Anyway, doesn't need to block this PR -- just something to think about.